modalc-lib/modal-vector.rkt

```diff
@@ -0,0 +1,45 @@
+#lang racket
+
+(require modalc)
+
+(define (modal-vector/c should-apply-ctc? . inner-ctc)
+  (define vector-contract (apply vector/c inner-ctc))
+  (define inner-ctc-proj
+    (contract-late-neg-projection vector-contract))
+  (make-contract
+   #:name `(modal-vector/c ,(contract-name vector-contract))
+   #:late-neg-projection
+   (λ (blame)
+     (define inner-ctc-proj/blame (inner-ctc-proj blame))
+     (λ (val neg-party)
+       (impersonate-vector val
+                           (λ (vec index ret)
+                             (if (should-apply-ctc? val)
```

Comment:
Now down here, I think the mode should probably receive both the index and the value, but not the vector -- i.e. `(list index ret)`